optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/incremental/IncrementalScoreCalculator.java

```diff
@@ -30,9 +30,10 @@
  * Any implementation is naturally stateful.
  *
  * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
+ * @param <Score_> the score type to go with the solution
  * @see IncrementalScoreDirector
  */
-public interface IncrementalScoreCalculator<Solution_> {
+public interface IncrementalScoreCalculator<Solution_, Score_ extends Score<Score_>> {
```

Comment:
For some reason, this is not public API. Yet, this change will affect users.

- Shouldn't we make it public API?
- Do we want to make this change?

My answer to both is a clear "yes".

Reply:
The reason this isn't public API is because it uses ScoreDirector. EasyScoreCalculator does not.
But let's discuss because that might not be enough reason...